core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -322,65 +349,59 @@ private[spark] class SparkIllegalArgumentException(
 private[spark] class SparkIndexOutOfBoundsException(
     errorClass: String,
     errorSubClass: Option[String] = None,
-    messageParameters: Array[String])
+    messageParameters: Map[String, String])
```

Comment:
The exception is used only once by the error class INDEX_OUT_OF_BOUNDS which is not tested. There is the JIRA https://issues.apache.org/jira/browse/SPARK-38734 for testing.